test/e2e/checkinterval_test.go

```diff
@@ -0,0 +1,99 @@
+	domain := name.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(name, domain)
+
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", name, err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, availableConditionsForPrivateIngressController...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	if err := kclient.Get(context.TODO(), name, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	}
+
+	routerDeployment, err := getDeployment(t, kclient, controller.RouterDeploymentName(ic), 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get router deployment: %v", err)
+	}
+
+	// set spec.tuningOptions.healthCheckInterval to a nondefault value, double the default
+	if err := setHealthCheckInterval(t, kclient, 10*time.Second, &metav1.Duration{2 * defaultHealthCheckInterval}, ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	if err := waitForIngressControllerCondition(t, kclient, 5*deploymentTimeout, name, availableConditionsForPrivateIngressController...); err != nil {
+		t.Fatalf("failed to observe expected conditions on %s: %v", name, err)
+	}
+
+	// verify the update to double the default
+	if err := waitForDeploymentEnvVar(t, kclient, routerDeployment, deploymentTimeout, ingresscontroller.RouterBackendCheckInterval, (2 * defaultHealthCheckInterval).String()); err != nil {
+		t.Fatalf("expected router deployment to set %s to %v: %v", ingresscontroller.RouterBackendCheckInterval, 2*defaultHealthCheckInterval, err)
+	}
+
+	// set spec.healthCheckInterval to an unacceptable value, 0s
+	if err := setHealthCheckInterval(t, kclient, 1*time.Minute, &metav1.Duration{0 * time.Second}, ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	if err := waitForIngressControllerCondition(t, kclient, 5*deploymentTimeout, name, availableConditionsForPrivateIngressController...); err != nil {
+		t.Fatalf("failed to observe expected conditions on %s: %v", name, err)
+	}
+
+	// when set to an unacceptable value, it should not be accepted
+	if err := waitForDeploymentEnvVar(t, kclient, routerDeployment, deploymentTimeout, ingresscontroller.RouterBackendCheckInterval, (0 * time.Second).String()); err == nil {
+		t.Fatalf("unexpected router deployment %s set at %v: %v", ingresscontroller.RouterBackendCheckInterval, 2*defaultHealthCheckInterval, err)
+	}
+
+	// when set to an unacceptable value, the env variable should not exist
+	if err := waitForDeploymentEnvVar(t, kclient, routerDeployment, deploymentTimeout, ingresscontroller.RouterBackendCheckInterval, ""); err != nil {
+		t.Fatalf("expected router deployment to remove %s: %v", ingresscontroller.RouterBackendCheckInterval, err)
+	}
```

Comment:
Are both `waitForDeploymentEnvVar` calls necessary?  It seems like the second one (make sure the environment variable is absent) would suffice, and the first one (make sure the environment variable isn't set to an unexpected value) is superfluous.

Reply:
ok